Service update for 1.4.365

mregen opened the pull request:
- Merge master with 1.4.365 release branch
- release JSON validator fix
- release improvements for cert validator
- otherwise test updates / comment improvements / latest nuget packages / ECDsa (only used in tests)

dependabot[bot] committed:
Bump System.ServiceModel.Primitives from 4.5.3 to 4.8.0 (#1182)

Bumps [System.ServiceModel.Primitives](https://github.com/dotnet/wcf) from 4.5.3 to 4.8.0.
- [Release notes](https://github.com/dotnet/wcf/releases)
- [Commits](https://github.com/dotnet/wcf/commits)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ThomasNehring committed:
Improve handling of certificate validation (#1217)

The change ensures that all certificate validation failures (except those related to the hostname of a server) are stacked in the right order into a service result, which -- if not empty -- in the end is raised in a service result exception.
The single point where knowledge about suppressible changes is needed is the place in Validate, where that exception is caught.

mregen committed:
Replace CRLF in debug strings with AppendLine and improve cert validator tests (#1220)

- improve cross platform support, replace CRLF by AppendLine 
- replace $format trace messages with {} style
- fix warnings for comments and protected fields, null parameter checks
- replace cert factory calls with new API (obsolete warnings)
- allow to run cert validator tests in parallel using new temp cert validator class
- more test cases for cert validator, e.g wth cert validation callback
- improve cert validator for special cases

dependabot[bot] committed:
Bump Portable.BouncyCastle from 1.8.8 to 1.8.9 (#1222)

* Bump Portable.BouncyCastle from 1.8.8 to 1.8.9

Bumps [Portable.BouncyCastle](https://github.com/novotnyllc/bc-csharp) from 1.8.8 to 1.8.9.
- [Release notes](https://github.com/novotnyllc/bc-csharp/releases)
- [Commits](https://github.com/novotnyllc/bc-csharp/compare/pcl-v1.8.8...pcl-v1.8.9)

Signed-off-by: dependabot[bot] <support@github.com>

* Update Opc.Ua.Core.Tests.csproj

remove unnecessary dependency

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Martin Regen <mregen@microsoft.com>

mregen committed:
Escape special characters in JSON fieldnames (#1224)

- Values were properly escaped, field names not.
- Fix fieldname encoding in various places
- Tests to verify strings are escaped in all codepaths.
- Fixes #1210

mregen committed:
Tests for ECDsa certs (#1225)

* finalizing some work on API surface for ECDsa certs
* yet only nist/brainpool certs are supported specifically for OPC UA ECC errata 
* split tests for ECDsa and RSA certs
* skip brainpool tests on macos, no OS support on macos 10.15

mregen committed:
Validate server domains in Cert validator (#1226)

- Based on new handling of multiple suppressible errors, add also the domain check to the list
- Move domain check in validator and use cert validation callback to handle BadCertificateInvalidHostname.
- Allow to handle all suppressible errors including the domain check by the cert validator callback, approve all errors by new AcceptAll flag. Applications which only set Approve are called for each of the suppressible errors and must approve eah of them.
- Add approval of multiple errors to sample UI
- fix a few warnings

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OPCFoundation/UA-.NETStandard?pullRequest=1229) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/OPCFoundation/UA-.NETStandard?pullRequest=1229) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: mregen<br/>:white_check_mark: ThomasNehring<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OPCFoundation/UA-.NETStandard?pullRequest=1229) it.</sub>

mregen committed:
Implement an AcceptAll flag in the cert validator callback (#1230)

- The AcceptAll flag can be used to accept all suppressible errors in one call back
- Behaviour unchanged when only Accept is set, callback is invoked for all errors and all must be approved

barnstee commented on Libraries/Opc.Ua.Gds.Server.Common/JsonApplicationsDatabase.cs:
```diff
@@ -54,17 +55,22 @@ public JsonApplicationsDatabase(string fileName)
             catch
             {
-                return new JsonApplicationsDatabase(fileName);
+
```
Please add something like "// nothing to do" so it doesn't look like a bug.
